src/server/object_services/object_server.js

```diff
@@ -1306,6 +1306,22 @@ function report_endpoint_problems(req) {
     }
 }
 
+/**
+ *
+ * READ_UPLOAD_MD
+ *
+ */
+async function read_upload_md(req) {
+    let obj;
+    load_bucket(req);
+    let res = {};
+    obj = await MDStore.instance().find_object_latest(req.bucket._id, req.rpc_params.key, true);
+    if (obj) {
+        res = { obj_id: obj._id, key: obj.key, xattr: obj.xattr };
+    }
+    return res;
```

Comment:
Why are we failing silently?
I wouldn't want this to be the normal API semantics, only if there's really a good reason for it.

Reply:
ok so we call that function in complete multipart upload (please notice that in init multipart upload we will store the upload md only if params.xattr exist - I did it that way because I wanted to save time and place complexity)  
we are failing silently here because of the fact that we don't know (in complete multipart upload) if there is MD for the object or not.

another solution is to always store the upload md and throw here error when needed instead of {}. do you prefer it ?

Reply:
So just catch the error in complete_multipart_upload() and ignore it if that's what needed.

Also I'm not really following the optimization and flow around xattr - how is xattr related to locating upload md?